data_structures/arrays/strings/almost_palindrome_checker.rb

```diff
@@ -0,0 +1,59 @@
+# Challenge name: Almost Palindrome
+#
+# Given a string s, return true if the s can be palindrome after deleting at most one character from it.
+#
+# Example 1:
+# Input: s = "aba"
+# Output: true
+#
+# Example 2:
+# Input: s = "abca"
+# Output: true
+# Explanation: You could delete the character 'c'.
+#
+# Example 3:
+# Input: s = "abc"
+# Output: false
+#
+# Constraints:
+# 1 <= s.length <= 105
+# s consists of lowercase English letters.
+
+# Approach 1: Two Pointers
+# Time Complexity: O(n)
```

Comment:
What's the space complexity of this algorithm?

Reply:
@jsca-kwok Even though we have recursion calls it won't get deeper than 2 levels so the space complexity is constant. Time complexity is linear because at most we will go through the list of characters once.